### PR TITLE
 c7n-left - value_from fix, env var interpolation support, and docs on data resources plus a tag test

### LINKED ISSUE #8879: docs - c7n-left example of traverse to data resource and policy on data resource.
the syntax is a little different for a policy on a data resource Ala `terraform.data.aws_ami` and then carries through sans terraform prefix with the traverse filter.

### LINKED ISSUE #8888: c7n-left - support value_from
currently it results in an exception if used.

```
  File "/src/c7n/filters/core.py", line 631, in match
    values = ValuesFrom(self.data['value_from'], self.manager)
  File "/src/c7n/resolver.py", line 128, in __init__
    'account_id': manager.config.account_id,
AttributeError: 'TerraformResourceManager' object has no attribute 'config'
```

### LINKED ISSUE #8889: c7n-left - allow using environment variables in policies
CI systems have rich information in their environment variables, we should allow their use in c7n-left policies such that the policy can customize itself based on the information therein.

### PR DESCRIPTION
**Pull request by kapilt**
closes #8889
closes #8888
closes #8879

### CODE REVIEW

File: tools/c7n_left/c7n_left/policy.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+def load_policies(policy_dir, options):
+    loader = DirectoryLoader(config=options)
+    loader.collection_class = LeftCollection
+    policies = loader.load_directory(policy_dir, recurse=False)
+    if not policies:
+        return ()
+
+    providers = {p.provider_name for p in policies}
+    assert len(providers), "only a single provider per policy dir"
```
Comment (johnsca) [inline tools/c7n_left/c7n_left/policy.py:19]: Message doesn't match the logic. Should this be `assert len(providers) == 1`?

**Comment by kapilt** [comment]
the doc build errors/warnings are weird

```
/home/runner/work/cloud-custodian/cloud-custodian/docs/source/tools/c7n-left.md:236: WARNING: Could not lex literal_block as "yaml". Highlighting skipped.
/home/runner/work/cloud-custodian/cloud-custodian/docs/source/tools/c7n-left.md:249: WARNING: Could not lex literal_block as "yaml". Highlighting skipped.
/home/runner/work/cloud-custodian/cloud-custodian/docs/source/tools/c7n-left.md:279: WARNING: Could not lex literal_block as "yaml". Highlighting skipped.
```

but pygments parses/renders them fine

Screenshot 2023-08-31 at 11 51 03 AM

File: tools/c7n_left/c7n_left/policy.py (hunk 1 above)
Comment (kapilt) [reply tools/c7n_left/c7n_left/policy.py]: thanks, fixed.

**Comment by kapilt** [comment]
apparently doc build issue is due to some tabs mixed in there :/, fix incoming.

**Pull request merged**